Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -384,13 +384,68 @@ extension SwiftLanguageServer {
           }
           let isInsertTextSnippet = clientSupportsSnippets && text != insertText
 
+          let textEdit: TextEdit?
+          if let text = text {
+            let utf8CodeUnitsToErase: Int = value[self.keys.num_bytes_to_erase] ?? 0
+
+            // Compute the TextEdit
+            if utf8CodeUnitsToErase == 0 {
+              // Nothing to delete. Fast path and avoid UTF-8/UTF-16 conversions
+              textEdit = TextEdit(range: completionPos..<req.params.position, newText: text)
+            } else if utf8CodeUnitsToErase == 1 {
+              // Fast path: Erasing a single UTF-8 byte code unit means we are also need to erase exactly one UTF-16 code unit, meaning we don't need to process the file contents
+              if completionPos.utf16index >= 1 {
+                let textEditRangeStart = Position(line: completionPos.line, utf16index: completionPos.utf16index - 1)
+                textEdit = TextEdit(range: textEditRangeStart..<req.params.position, newText: text)
+              } else {
+                // We need to delete past the start of the line. Ignore the replacement as below.
+                textEdit = TextEdit(range: completionPos..<req.params.position, newText: text)
+              }
+            } else {
+              // We need to delete characters. Let's compute the corresponding TextEdit
+              assert(completionPos.line == req.params.position.line)
+              // Construct a string index for the edit range start by subtracting the UTF-8 code units to erase from the completion position.
+              let line = snapshot.lineTable[completionPos.line]
+              let completionPosStringIndex = snapshot.lineTable.stringIndexOf(line: completionPos.line, utf16Column: completionPos.utf16index)!
+              let deletionStartStringIndex = line.utf8.index(completionPosStringIndex, offsetBy: -utf8CodeUnitsToErase)
+
+              // Compute the UTF-16 offset of the deletion start range. If the start lies in a previous line, this will be negative
+              let deletionStartUtf16Offset = line.utf16.distance(from: line.startIndex, to: deletionStartStringIndex)
+
+              // Check if we are only deleting on one line. LSP does not support deleting over multiple lines.
+              if deletionStartUtf16Offset >= 0 {
+                // We are only deleting characters on the same line. Construct the corresponding text edit.
+                let textEditRangeStart = Position(line: completionPos.line, utf16index: deletionStartUtf16Offset)
+                textEdit = TextEdit(range: textEditRangeStart..<req.params.position, newText: text)
+              } else {
+                // We are supposed to delete characters over multiple lines. This is not supported by LSP.
+                // Fall back to not replacing anything.
```

Comment:
s/not replacing anything/ignoring num_bytes_to_erase/

It still replaces the identifier characters in the range `completionPos..<req.params.position`